### PR TITLE
feat: [CDS-81225]: Added support for timeout for stage

### PR DESCRIPTION
**Pull request by pankajsingh-harness**
### Summary
This PR adds timeout support for the stage in the pipeline and template for following stages:

1. Deploy
2. Build
3. Approval
4. Security Test
5. Pipeline
6. Custom Stage

This can be seen in YAML: https://qa.harness.io/ng/account/px7xd_BFRCi-pfWPYXVjvw/all/orgs/default/projects/Pankaj/pipelines/Stage_Timeout/pipeline-studio?storeType=INLINE

**Execution Link:**

- https://qa.harness.io/ng/account/px7xd_BFRCi-pfWPYXVjvw/all/orgs/default/projects/Pankaj/pipelines/Stage_Timeout_Demo/executions/?storeType=INLINE
- https://qa.harness.io/ng/account/px7xd_BFRCi-pfWPYXVjvw/all/orgs/default/projects/Pankaj/pipelines/Deploy_Stage_Timeout/executions/?storeType=INLINE

#### Screenshots
**Feature demo:** https://drive.google.com/file/d/1W0M4YwjpKttG-g2RBwzZKSdYRyPOFaQB/view?usp=sharing

NA

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Sonar: `retrigger sonar`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress Rest: `retrigger cypress-rest`
- Cypress CD: `retrigger cypress-cd`
- Cypress Pipeline: `retrigger cypress-pipeline`
- Cypress CV: `retrigger cypress-cv`
- Fix Prettier: `fix prettier`
- MessageMetadata: `trigger messagecheck`
- GitLeaks: `trigger gitLeaks`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

### CODE REVIEW

**Comment by pankajsingh-harness** [comment]
retrigger titlecheck

File: src/modules/75-cd/components/PipelineStudio/StageTimeout/StageTimeout.tsx [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+  const { allowableTypes } = usePipelineContext()
+  const { expressions } = useVariablesExpression()
+  const { getString } = useStrings()
+
+  return (
+    <Formik<{ timeout?: string }>
+      initialValues={{
+        timeout: get(data, 'stage.timeout', '')
+      }}
+      onSubmit={/* istanbul ignore next */ () => undefined}
```
Comment (sunnykesh-harness) [inline src/modules/75-cd/components/PipelineStudio/StageTimeout/StageTimeout.tsx:42]: Is validation working without Submit handler? Also if possible plz remove Istambul ignore. Using noop we can save the need of placeholder function as well.

File: src/modules/75-cd/components/PipelineStudio/StageTimeout/StageTimeout.tsx [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+      validationSchema={Yup.object().shape({
+        timeout: getDurationValidationSchema({
+          minimum: '10s',
+          minimumErrorMessage: getString('validation.timeout10SecMinimum')
+        })
+      })}
+    >
+      {() => {
+        return (
+          <div className={cx(stepCss.formGroup, stepCss.sm)}>
```
Comment (sunnykesh-harness) [inline src/modules/75-cd/components/PipelineStudio/StageTimeout/StageTimeout.tsx:53]: In demo video saw separate input for default value, what if default is none of the allowed value?

File: src/modules/75-cd/components/PipelineStudio/StageTimeout/StageTimeout.tsx (hunk 1 above)
Comment (pankajsingh-harness) [reply src/modules/75-cd/components/PipelineStudio/StageTimeout/StageTimeout.tsx:42]: Yes @sunnykesh-harness. Validations has nothing do to with `onSubmit ` function. Let me remove Istanbul and add placeholder function

File: src/modules/75-cd/components/PipelineStudio/StageTimeout/StageTimeout.tsx (hunk 2 above)
Comment (pankajsingh-harness) [reply src/modules/75-cd/components/PipelineStudio/StageTimeout/StageTimeout.tsx:53]: @sunnykesh-harness the default value is from the runtime allowed values support.

File: src/modules/75-cd/components/PipelineStudio/StageTimeout/StageTimeout.tsx (hunk 1 above)
Comment (pankajsingh-harness) [reply src/modules/75-cd/components/PipelineStudio/StageTimeout/StageTimeout.tsx]: Done. @sunnykesh-harness please re-review.

**Review by sunnykesh-harness — APPROVED** [review]

**Review by rameshx — APPROVED** [review]

**Review by PratyushGar — APPROVED** [review]

**Review by vardanbansal-harness — APPROVED** [review]

**Review by samarth-sharma31 — APPROVED** [review]

**Pull request merged**